### DISCUSSION TITLE
get all organizations to which a user is granted to projects

### DISCUSSION THREAD

**Discussion by mbecker** [zitadel/zitadel#5229] [Q&A]
Hi,

I've followed and setup the B2B sample application as described here: [https://zitadel.com/docs/examples/login/nextjs-b2b](https://zitadel.com/docs/examples/login/nextjs-b2b)

In the header navigation of the NextJS-App you see a list with just one value of the user's "home organization" . My assumption would be that all organizations are listed to which a user is granted to the projects.

<img width="557" alt="Screenshot 2023-02-16 at 19 51 45" src="https://user-images.githubusercontent.com/26771/219460104-a2ffa23f-4044-4192-9e02-2a135d3a3c0a.png">

The documentation of the API [ListMyProjectOrgs](https://zitadel.com/docs/apis/proto/auth#listmyprojectorgs) defines as follows:

> Returns a list of organisations where the authorized user has a user grant (authorization) in the context of the requested project

Again, my assumption would be that all organizations are returned (to which the user is granted to the projects).

The result is as follows:
```json
{
    "details": {
        "totalResult": "1",
        "processedSequence": "207",
        "viewTimestamp": "2023-02-16T12:42:09.506173Z"
    },
    "result": [
        {
            "id": "201418996057112579",
            "details": {
                "sequence": "201",
                "creationDate": "2023-02-16T12:37:57.125284Z",
                "changeDate": "2023-02-16T12:37:57.125284Z",
                "resourceOwner": "201418996057112579"
            },
            "state": "ORG_STATE_ACTIVE",
            "name": "DemoCustomer",
            "primaryDomain": "democustomer.localhost"
        }
    ]
}
````

The return JSON contains only the user's "home organization".

Is it possible to get all organizations of the projects a user is granted to?

My use case is something like that:
A SaaS company has a public registration. The user A registers with an email address for a free plan. The user B registers with an email address for a free plan. Then later, user A decides to go premium which includes to have multiple user and roles. User A wants to invite user B. So, user B is granted access to user's A organization / project. In the SaaS portal user B can now switch between the two organizations.
Or, a company Y creates an organization, add it's Azure AD identity provider and wants to invite other users who are already signed up.

In the backend, a resource belongs to an organization. The authorization process decides based on the user and it's organiaztion / roles / ...

Would that be possible with Zitadel? Thanks for your help!

**Comment by mffap** [comment]
Hi @mbecker - that call you used will return organizations where the user has a manager role. So if you assign your user a manager role in another orgnanization, you should get back the full list.

On a separate note regarding you use case. I think you can achieve this with project grants. In the demo, if you give a user authorization on a granted project, it should appear in the demo app (we get the granted projects for an organization). You can have one project for the "base subscription" and grant that to the customer's organization. When upgrading you would grant an additional project and give the users the authorization to use it as well.
Would that make sense for your use case?
